app/internal/window/os_js.go

```diff
@@ -466,43 +475,59 @@ func (w *window) ShowTextInput(show bool) {
 // Close the window. Not implemented for js.
 func (w *window) Close() {}
 
+func (w *window) resize() {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	w.scale = float32(w.window.Get("devicePixelRatio").Float())
+
+	rect := w.cnv.Call("getBoundingClientRect")
+	w.size.X = float32(rect.Get("width").Float()) * w.scale
+	w.size.Y = float32(rect.Get("height").Float()) * w.scale
+
+	if vx, vy := w.visualViewport.Get("width"), w.visualViewport.Get("height"); !vx.IsUndefined() && !vy.IsUndefined() {
```

Comment:
Now that w.visualViewport is always set, is the IsUndefined tests still necessary?

Reply:
Yes. The `w.visualViewport` can be either "Window" or "Visual Viewport". The `w.window` doesn't have `window.width`/`window.height`.